src/MutableMap/MutableMapFeasability.dfy

```diff
@@ -0,0 +1,126 @@
+include "MutableMap.dfy"
+
+/*******************************************************************************
+*  Copyright by the contributors to the Dafny Project
+*  SPDX-License-Identifier: MIT
+*******************************************************************************/
+
+// RUN: %dafny /compile:0 "%s"
+
+module {:options "/functionSyntax:4"} MutableMapTrait {
+  trait MutableMapTrait<K(==),V(==)> {
```

Comment:
Love the feasibility implementation! There are ways of avoiding duplicating the specification without resorting to `{:termination false}` though. For example see what I did for the bootstrapping `Sequence<T>` trait here: https://github.com/dafny-lang/dafny/blob/master/Source/DafnyRuntime/DafnyRuntimeDafny/src/dafnyRuntime.dfy Namely, I have the actual trait in an abstract module that is refined per target language, and that means I can refine for Dafny as well.

Reply:
After an in-depth offline investigation, @stefan-aws and I decided that leaving the contract redundancy in for now is okay. Because the test case in this example aims to test the Java code from Dafny, the approach used in the Go runtime cited above doesn't quite work directly. There may be a way to do something similar to reduce the redundancy in the future, but we should be able to do so without changing the public API.